smartsim/_core/_install/builder.py

```diff
@@ -97,7 +100,34 @@ def from_str(cls, string: str, /) -> "Architecture":
 
 class Device(enum.Enum):
     CPU = "cpu"
-    GPU = "gpu"
+    CUDA118 = "cuda118"
+    CUDA121 = "cuda121"
+
+    @classmethod
+    def from_string(cls, str_: str) -> "Device":
+        str_ = str_.lower()
+        if str_ == "gpu":
+            # TODO: auto detect which device to use
+            #       currently hard coded to `cuda11`
+            return cls.CUDA118
+        return cls(str_)
+
+    def is_gpu(self) -> bool:
+        return self != type(self).CPU
+
+    def is_cuda(self) -> bool:
+        cls = type(self)
+        return self in (cls.CUDA118, cls.CUDA121)
+
+    def torch_suffix(self) -> str:
+        cls = type(self)
+        if self == cls.CPU:
```

Comment:
Thoughts on changing this into a dictionary definition instead of multiple `if`/`return`

Reply:
Totally reasonable to me! I originally had code like this in a couple of places wanted exhaustiveness checking from `_assert_never`, and I just kinda naively moved it over.

Now these values are sitting right next to their definition I don't see a huge need for the check anymore. Will remove!